### PR TITLE
Try using new MSON-like syntax for parameters

### PR DESCRIPTION
**Pull request by philsturgeon**
Doesn't seem to work locally, I get:

```
>> Line 203: ignoring unrecognized block (warning code 5)
```

Since I started this branch I noticed that Apiary itself also errors, so I guess there is some new MSON-lish approach to this.

I'll go ask them what is going on, then update this PR accordingly.

### CODE REVIEW

**Comment by danielgtaylor** [comment]
@philsturgeon thanks for working on updating the example! I've been reading through the [MSON specification](https://github.com/apiaryio/mson/blob/master/MSON%20Specification.md) and the [API Blueprint specification](https://github.com/apiaryio/api-blueprint/blob/master/API%20Blueprint%20Specification.md#def-uriparameters-section) and have come up with the following that works:

```md
## User List [/users{?name,joinedBefore,joinedAfter,sort,limit}]
A list of users

+ Parameters

    + name: `alice` (string, optional) - Search for a user by name
    + joinedBefore: `2011-01-01` (string, optional) - Search by join date
    + joinedAfter: `2011-01-01` (string, optional, ) - Search by join date
    + sort: `joined` (enum[string], optional) - Which field to sort by
        + Default: `name`
        + Members
            + `name`
            + `joined`
            + `-joined`
            + `age`
            + `-age`
            + `location`
            + `-location`
            + `plan`
            + `-plan`
    + limit: `25` (number, optional) ... The maximum number of users to return, up to `50`
        + Default: `10`
```

According to the spec, you must use `+ Members` (and don't use a `:` after it!). Also, I had to install the latest (0.19.x) version of Protagonist, otherwise it doesn't understand the new syntax. Does that fix the errors for you as well?

Also, please note that Aglio doesn't yet support MSON request/response bodies in the template yet. That's something that will be coming soon!

**Comment by danielgtaylor** [comment]
By the way, line 101 should probably be updated as well:

```md
    + id: `68a5sdf67` (required, string) - The note ID
```

**Comment by philsturgeon** [comment]
Oh nice find!

Let me explain WTF I'm doing.

I am trying to release http://apisyouwonthate.com/ in print form, and I am getting absolute hell from the section that shows off the syntax for declaring a property. Switching to MSON style url parameters fixed that because it's slightly shorter.

``` diff
-+ <param name> [= `<default>`] [([required | optional], [<type>], [`<example value>`])] [... <description>]
++ <parameter name>: `<example value>` (<type> | enum[<type>], required | optional) - <description>
```

Lol. 

So, that made me run around trying to update the whole f**king chapter for the sake of consistency. 

image

Your change to `Members` has worked! I updated my PR. 

I don't need full MSON support at this point, but I will revisit in a month or two. Could you merge this now so my book isn't full of lies when it comes out in a week, and I'll just keep an eye out for MSON support at a later date?

**Comment by zdne** [comment]
@danielgtaylor @philsturgeon a word of warning: URI parameters (section) isn't fully MSON compliant (yet), so please do not expect MSON to work there. We have just merely aligned the two syntaxes

File: example.md [hunk 1]
```diff
@@ -212,9 +212,9 @@ A list of users
             + `-location`
             + `plan`
             + `-plan`
-
-    + limit = `10` (optional, integer, `25`) ... The maximum number of users to return, up to `50`
-
+    + limit: `25` (integer, optional) ... The maximum number of users to return, up to `50`
```
Comment (danielgtaylor) [inline example.md]: Looks like this line still has the older `...` syntax!

**Comment by danielgtaylor** [comment]
I updated that last line and merged. Thanks for the PR :+1:

**Comment by philsturgeon** [comment]
Thanks for cleaning up my mess!

-- 
Phil

**Pull request merged**